quartz/plugins/transformers/lastmod.ts

```diff
@@ -1,58 +1,96 @@
+    )
+    return
+  }
+
+  const dt = [
+    // ISO 8601 format, e.g. "2024-09-09T00:00:00[Africa/Algiers]", "2024-09-09T00:00+01:00", "2024-09-09"
+    DateTime.fromISO,
+    // RFC 2822 (used in email & RSS) format, e.g. "Mon, 09 Sep 2024 00:00:00 +0100"
+    DateTime.fromRFC2822,
+    // Luxon is stricter about the format of the datetime string than `Date`
+    // fallback to `Date` constructor iff Luxon fails to parse datetime
+    (s: string, o: DateTimeOptions) => DateTime.fromJSDate(new Date(s), o),
+  ]
+    .values()
+    .map((f) => f(d, opts))
+    // find the first valid parse result
+    .find((dt) => dt != null && dt.isValid)
+
+  if (dt == null) {
     console.log(
       chalk.yellow(
         `\nWarning: found invalid date "${d}" in \`${fp}\`. Supported formats: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date#date_time_string_format`,
       ),
     )
+    return
   }
-
-  return invalidDate ? new Date() : dt
+  return dt
 }
 
-type MaybeDate = undefined | string | number
 export const CreatedModifiedDate: QuartzTransformerPlugin<Partial<Options>> = (userOpts) => {
   const opts = { ...defaultOptions, ...userOpts }
+  const parseOpts: DateTimeOptions = {
+    setZone: true,
+    zone: opts.defaultTimezone,
+  }
   return {
     name: "CreatedModifiedDate",
     markdownPlugins() {
       return [
         () => {
           let repo: Repository | undefined = undefined
           return async (_tree, file) => {
-            let created: MaybeDate = undefined
-            let modified: MaybeDate = undefined
-            let published: MaybeDate = undefined
+            let created: DateTime | undefined = undefined
+            let modified: DateTime | undefined = undefined
+            let published: DateTime | undefined = undefined
 
             const fp = file.data.filePath!
             const fullFp = path.isAbsolute(fp) ? fp : path.posix.join(file.cwd, fp)
             for (const source of opts.priority) {
               if (source === "filesystem") {
                 const st = await fs.promises.stat(fullFp)
-                created ||= st.birthtimeMs
-                modified ||= st.mtimeMs
+                // birthtime can be 0 on some filesystems, so default to the earlier of ctime/mtime
+                created ||= DateTime.fromMillis(st.birthtimeMs || Math.min(st.ctimeMs, st.mtimeMs))
+                modified ||= DateTime.fromMillis(st.mtimeMs)
               } else if (source === "frontmatter" && file.data.frontmatter) {
-                created ||= file.data.frontmatter.date as MaybeDate
-                modified ||= file.data.frontmatter.lastmod as MaybeDate
-                modified ||= file.data.frontmatter.updated as MaybeDate
-                modified ||= file.data.frontmatter["last-modified"] as MaybeDate
-                published ||= file.data.frontmatter.publishDate as MaybeDate
+                created ||= parseDateString(fp, file.data.frontmatter.date, parseOpts)
```

Comment:
The datetime properties (created/modified/published) will be dealiased in the FrontMatter plugin (per #1664). Since we're codifying that those properties are datetimes, and frontmatter is the only source that uses string values, does it make sense to do the parsing in frontmatter instead?  

i.e.

```diff
diff --git a/quartz/plugins/transformers/frontmatter.ts b/quartz/plugins/transformers/frontmatter.ts
index d444b00..8b9d361 100644
--- a/quartz/plugins/transformers/frontmatter.ts
+++ b/quartz/plugins/transformers/frontmatter.ts
@@ -74,16 +74,16 @@ export const FrontMatter: QuartzTransformerPlugin<Partial<Options>> = (userOpts)
             const socialImage = coalesceAliases(data, ["socialImage", "image", "cover"])
 
             const created = coalesceAliases(data, ["date", "created"])
-            if (created) data.created = created
+            if (created) data.created = parseDateString(created, parseOpts)
             const modified = coalesceAliases(data, [
               "lastmod",
               "updated",
               "last-modified",
               "modified",
             ])
-            if (modified) data.modified = modified
+            if (modified) data.modified = parseDateString(modified, parseOpts)
             const published = coalesceAliases(data, ["publishDate", "published", "date"])
-            if (published) data.published = published
+            if (published) data.published = parseDateString(published, parseOpts)
 
             if (socialImage) data.socialImage = socialImage
 
@@ -103,9 +103,9 @@ declare module "vfile" {
     } & Partial<{
         tags: string[]
         aliases: string[]
-        modified: string
-        created: string
-        published: string
+        modified: DateTime
+        created: DateTime
+        published: DateTime
         description: string
         publish: boolean | string
         draft: boolean | string
```

Reply:
Yeah I think this makes sense. Actually you can cherrypick that PR into here (I can close that one)

Reply:
the other PR has been merged. Feel free to continue the work here.